core/src/main/java/com/graphhopper/routing/InstructionsFromEdges.java

```diff
@@ -196,19 +204,42 @@ public void next(EdgeIteratorState edge, int index, int prevEdgeId) {
                     .setDirOfRotation(deltaOut)
                     .setExited();
 
+            doublePrevName = prevName;
             prevName = name;
             prevAnnotation = annotation;
 
         } else {
             int sign = getTurn(edge, baseNode, prevNode, adjNode, annotation, name);
 
             if (sign != Instruction.IGNORE) {
```

Comment:
Ok, moving into a new package makes sense, will create a new issue for this. Not sure about the package protected, as I think this is needed for the clients as well?